### ISSUE TITLE
Question about CPR

### ISSUE THREAD

**Issue by suhasghorp** [lassoregression/mbs-val#1]
Thank you for making mbs-val available on Github, it's very useful for a student like me. You had mentioned that you recieved some data from Bloomberg and I was trying to figure out where that data is used in the code.

`cpr = 0.046  # Adjusted based on the screenshot data`

Is this value of CPR directly from Bloomberg for this particular pool? 

Thanks again.

**Comment by mujeeb-k** [comment]
Hey @suhasghorp ,

I'm glad to hear you find `mbs-val` useful. As a fellow student, I appreciate your interest.

Regarding your question about the CPR val:

The line:
```python
cpr = 0.046 # Adjusted based on the screenshot data
```
is based on general trends from Bloomberg data, not directly from Bloomberg for this specific pool.
This adjustment ensures realistic prepayment modeling in our analysis.

and the mortgage details:

```python
# Mortgage-backed security details
coupon_rate = 0.045  # Annual coupon rate from the screenshot
initial_principal = 854347  # Initial principal from the screenshot
current_principal = 625677  # Current principal from the screenshot
remaining_months = 360 - 61  # 61 months have passed since the origination
```

were derived from Bloomberg.

I hope this helps. Let me know if you have further questions.

**Comment by suhasghorp** [comment]
Awesome! Thanks! The interest rate paths are used to discount the cash flows, correct? My professor wants us to use the future interest rate paths to derive CPR itself and to discount of course. so these are what they call "short rates" and we need to convert these to a 10 Year swap rate (apparently these is a formula) and then determine the CPR based on that swap rate. It gets complicated ;] Thanks again.

**Comment by mujeeb-k** [comment]
Yes, you're correct. The interest rate paths are used to discount the cash flows.

Your professor's exercise sounds interesting. If you can, please share your work with me when you’re done.
